pyomo/core/plugins/transform/deactivate_trivial_constraints.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: UTF-8 -*-
+"""Transformation to deactivate trivial constraints."""
+import textwrap
+from math import fabs
+
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.component_set import ComponentSet
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class TrivialConstraintDeactivator(IsomorphicTransformation):
+    """Deactivates trivial constraints of form constant = constant."""
+
+    alias('core.deactivate_trivial_constraints',
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(TrivialConstraintDeactivator, self).__init__()
+        self.tolerance = 1E-13
+
```

Comment:
I never like seeing tolerance constants in code like this, even though I do it myself.  Would it make sense to have a global tolerance in something like pyomo.common (which @jsiirola is drafting to store common information/methods)?

Reply:
@whart222: In general, I agree.  That said, I am not sure it should hold up this PR.

A long time ago we started thinking about moving to a ConfigBlock-based approach for handling embedded constants like this.  I'd love to set up an issue to start reviving that design idea.

Reply:
I agree that this shouldn't hold up the PR.